### PR TITLE
check Content-Type from response

### PR DESCRIPTION
**Pull request by xiekeyang**
Signed-off-by: xiekeyang <xiekeyang@huawei.com>

### CODE REVIEW

File: tools/discovery/discovery.go [hunk 1]
```diff
@@ -89,6 +89,10 @@ func refEnginesFetching(parsedName map[string]string) (uri *url.URL, engines []e
 		return uri, nil, fmt.Errorf("ref engine fetching error, status code = %d", resp.StatusCode)
 	}
 
+	if resp.Header.Get(`Content-Type`) != `application/vnd.oci.ref-engines.v1+json; charset=utf-8` {
```
Comment (wking) [inline tools/discovery/discovery.go]: We can accept other charsets, although Go's JSON parser may be limited to [these][1].  Maybe splo=it on the optipnal semicolon to remove the charset portion (if present).

[1]: https://tools.ietf.org/html/rfc7159#section-8.1

Comment (wking) [reply tools/discovery/discovery.go]: > Maybe split on the optional semicolon…

The Go for this seems to be [`ParseMediaType`][1].

[1]: https://golang.org/pkg/mime/#ParseMediaType

**Comment by xiekeyang** [comment]
Updated

**Comment by wking** [comment]
This branch still needs to be deleted.  It looks like the `go-ref-engine-discovery-api` and `refact-api` branches could be deleted too.

**Pull request merged**